packages/telemetry/src/context-aware-slog.js

```diff
@@ -0,0 +1,475 @@
+        triggerContext = {
+          'run.id': `bootstrap-${finalBody.blockTime}`,
+          'run.trigger.time': finalBody.blockTime,
+          'run.trigger.type': 'bootstrap',
+        };
+        break;
+      }
+      case SLOG_TYPES.SWINGSET.BRIDGE_INBOUND:
+      case SLOG_TYPES.SWINGSET.DELIVER_INBOUND: {
+        const [blockHeight, txHash, msgIdx] = (
+          finalBody.inboundNum || ''
+        ).split('-');
+        const triggerType = body.type.split('-')[2];
+
+        triggerContext = {
+          'run.id': `${triggerType}-${finalBody.inboundNum}`,
+          'run.trigger.blockHeight': Number(blockHeight),
+          'run.trigger.msgIdx': Number(msgIdx),
+          'run.trigger.sender': finalBody.sender,
+          'run.trigger.source': finalBody.source,
+          'run.trigger.time': finalBody.blockTime,
+          'run.trigger.txHash': txHash,
+          'run.trigger.type': triggerType,
+        };
+        persistContext(triggerContext);
+        break;
+      }
+      case SLOG_TYPES.SWINGSET.COMMIT.FINISH:
+      case SLOG_TYPES.SWINGSET.COMMIT.START:
+      case SLOG_TYPES.SWINGSET.END_BLOCK.FINISH: {
+        assert(!!blockContext);
+        break;
+      }
+      case SLOG_TYPES.SWINGSET.END_BLOCK.START: {
+        assert(!!blockContext);
+        break;
+      }
+      case SLOG_TYPES.SYSCALL:
+      case SLOG_TYPES.SYSCALL_RESULT: {
+        logAttributes['crank.syscallNum'] = finalBody.syscallNum;
+
+        delete finalBody.deliveryNum;
+        delete finalBody.replay;
+        delete finalBody.syscallNum;
+
+        break;
+      }
+      default:
+        break;
+    }
+
+    /** @type {LogAttributes} */
+    logAttributes = {
+      ...blockContext,
+      ...crankContext,
+      ...initContext,
+      ...logAttributes,
+      ...replayContext,
+      ...triggerContext,
+    };
+
+    logger.emit({
+      attributes: JSON.parse(stringify(logAttributes)),
+      body: JSON.parse(stringify(finalBody)),
+      severityNumber: SeverityNumber.INFO,
+    });
+
+    /**
+     * Add any after report operations here
+     * like resetting context data
+     */
+    switch (body.type) {
+      case SLOG_TYPES.CRANK.RESULT: {
+        crankContext = null;
+        break;
+      }
+      case SLOG_TYPES.KERNEL.INIT.FINISH: {
+        initContext = null;
+        break;
+      }
+      case SLOG_TYPES.REPLAY.FINISH: {
+        replayContext = null;
+        break;
+      }
+      // eslint-disable-next-line no-restricted-syntax
+      case SLOG_TYPES.RUN.FINISH: {
+        triggerContext = null;
+        break;
+      }
+      case SLOG_TYPES.SWINGSET.AFTER_COMMIT_STATS: {
+        blockContext = null;
+        break;
+      }
+      case SLOG_TYPES.SWINGSET.BOOTSTRAP_BLOCK.FINISH: {
+        blockContext = null;
+        break;
+      }
+      case SLOG_TYPES.SWINGSET.END_BLOCK.START: {
+        triggerContext = restoreContext();
+        break;
+      }
+      default:
+        break;
+    }
+  };
+
+  return slogSender;
```

Comment:
I believe the otel SDK does some caching. Let's try to flush things when possible. For example here is what the otel tracing sender does: https://github.com/Agoric/agoric-sdk/blob/696d9370516dfe39e3f155d6aa4252d605dbaa6e/packages/telemetry/src/otel-trace.js#L57-L70

Reply:
Added flush and shutdown functions